File: jpa/JPA1/demo/domain/Delivery.java
package studyJPA.demo.domain;

import javax.persistence.*;

@Entity
public class Delivery {
    @Id @GeneratedValue
    @Column(name="delivery_id")
    private Long id;

    @OneToOne(mappedBy = "delivery", fetch = FetchType.LAZY)
    private Order order;

    @Embedded
    private Address address;

    @Enumerated(EnumType.STRING)
    private DeliverStatus status; // READY, COMP
}

File: kotlin/querydsl/src/main/kotlin/com/querydsl/querydsl/domain/Delivery.kt
package com.querydsl.querydsl.domain

import com.fasterxml.jackson.annotation.JsonIgnore
import lombok.Getter
import lombok.Setter
import javax.persistence.*

@Entity
@Getter
@Setter
@Table(name="delivery")
class Delivery {
    @Id @GeneratedValue
    @Column(name="delivery_id")
    var id : Long = 0;

    @JsonIgnore
    @OneToOne(mappedBy = "delivery", fetch = FetchType.LAZY)
    lateinit var order: Order

    @Embedded
    lateinit var address: Address ;

    @Enumerated(EnumType.STRING)
    lateinit var  status:DeliverStatus
// READY, COMP
}